### PR TITLE
Add onnx runtime text classification, token classification and question answering examples

### PR DESCRIPTION
**Pull request by echarlaix**
In this PR we add the ONNX Runtime examples for static and dynamic quantization for the following tasks : text classification, token classification and question answering. We also introduce a temporary `ORTModel` class which allows to evaluate an ORT model, this is useful in order to compare the evaluation results of the original full-precision model with the resulting quantized model.

### CODE REVIEW

File: examples/onnxruntime/pytorch/question-answering/run_qa.py [hunk 1]
```diff
@@ -0,0 +1,735 @@
...
+        quantizer.fit(
+            model_args.model_name_or_path,
+            output_dir,
+            feature="question-answering",
+            cache_dir=model_args.cache_dir,
+        )
+        onnx_config = quantizer.onnx_config
+        opt_model_path = output_dir.joinpath("model-quantized.onnx")
+
+    elif optim_args.optimize:
```
Comment (JingyaHuang) [inline examples/onnxruntime/pytorch/question-answering/run_qa.py:707]: Is it possible to apply quantization as well as optimization?

File: examples/onnxruntime/pytorch/question-answering/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<!---
+Copyright 2020 The HuggingFace Team. All rights reserved.
```
Comment (JingyaHuang) [inline examples/onnxruntime/pytorch/question-answering/README.md:2]: Is it necessary to change the year of license copyright?

File: examples/onnxruntime/pytorch/question-answering/run_qa.py [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+        )
+
+    # Preprocessing the datasets.
+    # Preprocessing is slighlty different for training and evaluation.
+    if training_args.do_train:
+        column_names = raw_datasets["train"].column_names
+    elif training_args.do_eval:
+        column_names = raw_datasets["validation"].column_names
+    else:
+        raise ValueError("--do_train or --do_eval are both set to False")
```
[the remark this answers is not in the document]
Comment (JingyaHuang) [reply examples/onnxruntime/pytorch/question-answering/run_qa.py:420]: We might assume that users will normally not come with an already fine-tuned model and just run the script for prediction.

File: examples/onnxruntime/pytorch/text-classification/run_glue.py [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+        quantizer.fit(
+            model_args.model_name_or_path,
+            output_dir,
+            feature="sequence-classification",
+            cache_dir=model_args.cache_dir,
+        )
+        onnx_config = quantizer.onnx_config
+        opt_model_path = output_dir.joinpath("model-quantized.onnx")
+
+    elif optim_args.optimize:
```
Comment (JingyaHuang) [inline examples/onnxruntime/pytorch/text-classification/run_glue.py:611]: Idem question on the possibility of quantization+optimization

File: examples/onnxruntime/pytorch/token-classification/run_ner.py [hunk 5]
```diff
@@ -0,0 +1,682 @@
...
+        quantizer.fit(
+            model_args.model_name_or_path,
+            output_dir,
+            feature="token-classification",
+            cache_dir=model_args.cache_dir,
+        )
+        onnx_config = quantizer.onnx_config
+        opt_model_path = output_dir.joinpath("model-quantized.onnx")
+
+    elif optim_args.optimize:
```
Comment (JingyaHuang) [inline examples/onnxruntime/pytorch/token-classification/run_ner.py:655]: Idem question on the possibility of quantization+optimization

File: optimum/onnxruntime/model.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
+#  Copyright 2021 The HuggingFace Team. All rights reserved.
```
Comment (JingyaHuang) [inline optimum/onnxruntime/model.py:1]: ORTModel has some overlaps with ORTTrainer, we might go for it in this PR and then merge it with ORTTrainer in PR#77!

**Review by JingyaHuang — COMMENTED** [review]
Nice work! 
Now we have examples well arranged with quantizer and optimizer, next, we could expand the examples with ORTTrainer implemented in PR#77.

File: examples/onnxruntime/pytorch/question-answering/run_qa.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (echarlaix) [reply examples/onnxruntime/pytorch/question-answering/run_qa.py:419]: You are right, I added the prediction step for all the examples scripts

File: examples/onnxruntime/pytorch/question-answering/run_qa.py (hunk 1 above)
Comment (echarlaix) [reply examples/onnxruntime/pytorch/question-answering/run_qa.py:729]: Optimization can be applied with quantization (when `optimize_model` attribute of `ORTConfig` is set to True), however, `ORTQuantizer` and `ORTOptimizer` are not very compatible (in the sense that if we first use `ORTOptimizer`  and then `ORTQuantizer` , a lot of operators will not be quantized)

**Pull request merged**